docs(migration): add general overview of API and tools changes in 8.8 (#6328)
camunda/camunda-docs: docs/apis-tools/migration-manuals/index.md

## API and SDK status

:::note
Start migration early to reduce upgrade risk beyond 8.8.
:::

For more information, see [Upcoming API Changes in Camunda 8: A Unified and Streamlined Experience](https://camunda.com/blog/2024/12/api-changes-in-camunda-8-a-unified-and-streamlined-experience/).

| Component / Use          | Status in 8.8  | Migrate to                 | Migrate by          |
| ------------------------ | -------------- | -------------------------- | ------------------- |
| V1 component APIs        | **Deprecated** | Orchestration Cluster API  | Before Camunda 8.10 |
| Spring Zeebe SDK         | **Deprecated** | Camunda Spring SDK         | Before Camunda 8.10 |
| Zeebe Process Test (ZPT) | **Deprecated** | Camunda Process Test (CPT) | Before Camunda 8.10 |
| Job-based user tasks     | **Deprecated** | Camunda User Tasks         | Before Camunda 8.10 |
| ZeebeClient              | **Deprecated** | Camunda Java Client        | Before Camunda 8.10 |
| Tasklist GraphQL API     | **Removed**    | Orchestration Cluster API  | 8.8                 |

### Orchestration Cluster API

> The Orchestration Cluster API replaces the legacy V1 component APIs, providing a unified interface for managing and interacting with the orchestration cluster. This API is designed to be more consistent and easier to use, aligning with the new architecture introduced in Camunda 8.8.

Reviewer: ```suggestion
The Orchestration Cluster API replaces the deprecated V1 component APIs, providing a unified interface for managing and interacting with the orchestration cluster. This API is designed to be more powerful in functionality and easier to use, aligning with the new architecture introduced in Camunda 8.8.
```

### Camunda Process Test (CPT)

The Camunda Process Test (CPT) framework is the successor to Zeebe Process Test (ZPT). It provides a more streamlined and efficient way to test process models in Camunda 8.8.
It is designed to work seamlessly with the new orchestration cluster architecture and offers improved performance and usability.
ZPT is officially deprecated in Camunda 8.8, and users are encouraged to migrate to CPT.

Timeline:

- **8.8 release:** Introduction of Camunda Process Test (CPT) as the successor to Zeebe Process Test (ZPT).
- **8.9 release:** Continued support for both ZPT and CPT.
- **8.10 release:** ZPT will be removed completely; customers must have migrated their tests to CPT by this time.

For more information, see the update guide <!-- _Link to the update guide._ -->

### Camunda User Tasks

The Camunda User Tasks feature replaces the legacy job-based user tasks in Camunda 8.8. This new feature provides a more robust and flexible way to handle user tasks within process models.
It is designed to work seamlessly with the new orchestration cluster architecture and offers improved performance and usability.

> Job-based user tasks are deprecated in Camunda 8.8, and users are encouraged to migrate to the new Camunda User Tasks feature.

Reviewer: ```suggestion
Job-based user tasks are deprecated in Camunda 8.8, and users are encouraged to migrate to the new Camunda User Tasks feature. The search functionality for job-based user tasks will be removed with 8.10.
```

For more information, see the [update guide](migrate-to-camunda-user-tasks.md)

### Camunda Java Client

> The Camunda Java Client is set to supersede the Zeebe Java Client, and there is a grace period for the Zeebe Client that will extend until version 8.10.

Reviewer: ```suggestion
The Camunda Java Client is a drop-in replacement for Zeebe Java Client, however, Zeebe Java Client is still available. Zeebe Java Client is deprecated with 8.8 and will be removed with 8.10.

> This means that while the Zeebe Client is deprecated, it will still be available for use until that version. In this way users can transition to the Camunda Java Client without immediate pressure, allowing for a smoother migration process.

Reviewer: ```suggestion
In this way users can transition to the Camunda Java Client without immediate pressure, allowing for a smoother migration process.
```

For more information, see the update guide <!-- _Link to the update guide._ -->

### Tasklist GraphQL API

> The Tasklist GraphQL API has been removed in Camunda 8.8. This change is part of the broader architectural evolution towards the Orchestration Cluster API, which provides a more unified and consistent interface for managing tasks and workflows.

Reviewer: ```suggestion
The previously deprecated Tasklist GraphQL API has been removed in Camunda 8.8. This change is part of the broader architectural evolution towards the Orchestration Cluster API, which provides a more unified and consistent interface for managing tasks and workflows.
```

> Check the User Task API documentation for more information on how to manage user tasks in Camunda 8.8.

Reviewer: ```suggestion
Check the Orchestration Cluster API on User Tasks for more information on how to manage user tasks in Camunda 8.8.
```
